drivers/display/display_gc9x01x.c

```diff
@@ -0,0 +1,745 @@
+/**
+ * Copyright (c) 2023 Mr Beam Lasers GmbH.
+ * Copyright (c) 2023 Amrith Venkat Kesavamoorthi <amrith@mr-beam.org>
+ * Copyright (c) 2023 Martin Kiepfer <mrmarteng@teleschirm.org>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "display_gc9x01x.h"
+
+#include <zephyr/dt-bindings/display/gc9x01x.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(display_gc9x01x, CONFIG_DISPLAY_LOG_LEVEL);
+
+/* Command/data GPIO level for commands. */
+#define GC9X01X_GPIO_LEVEL_CMD  0U
+/* Command/data GPIO level for data. */
+#define GC9X01X_GPIO_LEVEL_DATA 1U
+
+/* Maximum number of default init registers  */
+#define GC9X01X_NUM_DEFAULT_INIT_REGS (12U)
+
+/* Display data struct */
+struct gc9x01x_data {
+	uint8_t bytes_per_pixel;
+	enum display_pixel_format pixel_format;
+	enum display_orientation orientation;
+};
+
+/* Configuration data struct.*/
+struct gc9x01x_config {
+	struct spi_dt_spec spi;
+	struct gpio_dt_spec cmd_data;
+	struct gpio_dt_spec reset;
+	struct pwm_dt_spec backlight;
+	uint8_t pixel_format;
+	uint16_t rotation;
+	uint16_t x_resolution;
+	uint16_t y_resolution;
+	bool inversion;
+	const void *regs;
+	int (*regs_init_fn)(const struct device *dev);
```

Comment:
is this needed? future proofing? If there's no plan to use it just drop it, can always add it back later

Reply:
That was left over from a previous version.
Fixed in. [3af98ab](https://github.com/zephyrproject-rtos/zephyr/commit/3af98ab51e4a42df60a3209ef92480cd397189a3)